Error when start theme watch

I have an error when i run theme watch:
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x4721cc]

goroutine 1 [running]:
panic(0x781700, 0xc042006080)
        /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1af
github.com/Shopify/themekit/kit.(*FileWatcher).watchDirectory.func1(0xc04237c6c0, 0x10f, 0x0, 0x0, 0x9777a0, 0xc042265020, 0x0, 0x0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/file_watcher.go:68 +0x5c
path/filepath.walk(0xc0421eea20, 0x101, 0x97d2a0, 0xc04233f140, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:372 +0x235
path/filepath.walk(0xc042376400, 0xf5, 0x97d2a0, 0xc04233f020, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421c9680, 0xe8, 0x97d2a0, 0xc04233ec60, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042344fc0, 0xdf, 0x97d2a0, 0xc04233eb40, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042341880, 0xd2, 0x97d2a0, 0xc04233e720, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0420471e0, 0xc7, 0x97d2a0, 0xc042323e60, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc04232f380, 0xba, 0x97d2a0, 0xc042323b00, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc04232d550, 0xab, 0x97d2a0, 0xc0423238c0, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042326140, 0x9e, 0x97d2a0, 0xc042323560, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc04231ae60, 0x95, 0x97d2a0, 0xc042322180, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421eb5f0, 0x88, 0x97d2a0, 0xc0421e3e00, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421c5c80, 0x7c, 0x97d2a0, 0xc0421cec60, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421100e0, 0x6f, 0x97d2a0, 0xc042216660, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421c2930, 0x69, 0x97d2a0, 0xc0421bef00, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0421be3c0, 0x5c, 0x97d2a0, 0xc0421be420, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0423654a0, 0x51, 0x97d2a0, 0xc042365500, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042275770, 0x44, 0x97d2a0, 0xc04202f860, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042042200, 0x3a, 0x97d2a0, 0xc04202e6c0, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0423681b0, 0x2d, 0x97d2a0, 0xc042173aa0, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc042368db0, 0x23, 0x97d2a0, 0xc042367f20, 0xc0420c1960, 0x0, 0x0)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.walk(0xc0420c14e0, 0x16, 0x97d2a0, 0xc0420390e0, 0xc0420c1960, 0x0, 0x1)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:376 +0x34b
path/filepath.Walk(0xc0420c14e0, 0x16, 0xc0420c1960, 0x16, 0xc0420898e8)
        /usr/local/Cellar/go/1.7.1/libexec/src/path/filepath/path.go:398 +0xdc
github.com/Shopify/themekit/kit.(*FileWatcher).watchDirectory(0xc042104540, 0xc0420c14e0, 0x16, 0x0, 0x0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/file_watcher.go:78 +0xc6
github.com/Shopify/themekit/kit.newFileWatcher(0xc04224c3c0, 0xc04213bf10, 0xc04225ffa0, 0x17, 0x99a1c0, 0xc, 0xc, 0xc042268200, 0x3, 0x4, ...)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/file_watcher.go:62 +0x207
github.com/Shopify/themekit/kit.ThemeClient.NewFileWatcher(0xc04224c3c0, 0xc04213bf10, 0xc04225ffa0, 0x17, 0x99a1c0, 0xc, 0xc, 0xc042268200, 0x3, 0x4, ...)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/theme_client.go:44 +0xc2
github.com/Shopify/themekit/cmd.watch(0xc04213fe00, 0x1, 0x1, 0x0, 0x0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/cmd/watch.go:43 +0x3aa
github.com/Shopify/themekit/cmd.glob..func6(0x9970e0, 0x9b6c38, 0x0, 0x0, 0x0, 0x0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/cmd/watch.go:28 +0x59
github.com/Shopify/themekit/vendor/github.com/spf13/cobra.(*Command).execute(0x9970e0, 0x9b6c38, 0x0, 0x0, 0x9970e0, 0x9b6c38)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/vendor/github.com/spf13/cobra/command.go:599 +0x23b
github.com/Shopify/themekit/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x996860, 0x996a80, 0x995dc0, 0x0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/vendor/github.com/spf13/cobra/command.go:689 +0x36e
github.com/Shopify/themekit/vendor/github.com/spf13/cobra.(*Command).Execute(0x996860, 0x0, 0xc0420389c0)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/vendor/github.com/spf13/cobra/command.go:648 +0x32
main.main()
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/cmd/theme/main.go:9 +0x38

--- a/kit/asset.go
+++ b/kit/asset.go
@@ -95,7 +95,8 @@ func loadAssetsFromDirectory(dir string, ignore func(path string) bool) ([]Asset
 
 func loadAsset(root, filename string) (asset Asset, err error) {
 	asset = Asset{}
-	path := toSlash(root, filename)
+	path := filepath.ToSlash(filepath.Join(root, filename))
+	path = strings.Replace(path, "\\", "/", -1)
 	file, err := os.Open(path)
 	if err != nil {
 		return asset, fmt.Errorf("loadAsset: %s", err)
@@ -116,21 +117,13 @@ func loadAsset(root, filename string) (asset Asset, err error) {
 		return asset, fmt.Errorf("loadAsset: %s", err)
 	}
 
-	asset = Asset{Key: toSlash(filename)}
+	asset = Asset{Key: filepath.ToSlash(filename)}
 	if contentTypeFor(buffer) == "text" {
 		asset.Value = string(buffer)
 	} else {
 		asset.Attachment = base64.StdEncoding.EncodeToString(buffer)
 	}
 	return asset, nil
-}
-
-func toSlash(path ...string) string {
-	newpath := filepath.ToSlash(filepath.Join(path...))
-	if strings.Index(newpath, "\\") >= 0 {
-		newpath = strings.Replace(newpath, "\\", "/", -1)
-	}
-	return newpath
 }
 
 func contentTypeFor(data []byte) string {

--- a/kit/asset_test.go
+++ b/kit/asset_test.go
@@ -2,7 +2,6 @@ package kit
 
 import (
 	"sort"
-	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -87,7 +86,8 @@ func (s *LoadAssetSuite) TestLoadAssetsFromDirectory() {
 }
 
 func (s *LoadAssetSuite) TestLoadAsset() {
-	windowsRoot := strings.Replace("../fixtures/project", "/", "\\", -1)
+	windowsRoot := "..\\fixtures\\project"
+	println(windowsRoot)
 	asset, err := loadAsset(windowsRoot, "whatever.txt")
 	assert.Equal(s.T(), "whatever.txt", asset.Key)
 	assert.Equal(s.T(), true, asset.IsValid())

--- a/kit/file_watcher.go
+++ b/kit/file_watcher.go
@@ -62,15 +62,16 @@ func newFileWatcher(client ThemeClient, dir, notifyFile string, recur bool, filt
 	return newWatcher, newWatcher.watchDirectory(dir)
 }
 
-func (watcher *FileWatcher) watchDirectory(dir string) error {
-	dir = filepath.Clean(dir)
-	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if info.IsDir() && !watcher.filter.matchesFilter(path) && path != dir {
+func (watcher *FileWatcher) watchDirectory(root string) error {
+	root = filepath.Clean(root)
+	return filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() && !watcher.filter.matchesFilter(path) && path != root {
 			for _, dir := range assetLocations {
-				if strings.HasPrefix(path, dir) {
+				if strings.HasPrefix(path, filepath.Join(root, dir, string(filepath.Separator))) {
 					if err := watcher.watcher.Add(path); err != nil {
 						return fmt.Errorf("Could not watch directory %s: %s", path, err)
 					}
+					break
 				}
 			}
 		}

--- a/kit/file_watcher.go
+++ b/kit/file_watcher.go
@@ -175,14 +175,11 @@ func handleEvent(watcher *FileWatcher, event fsnotify.Event) {
 }
 
 func extractAssetKey(filename string) string {
-	filename = filepath.ToSlash(filename)
-
 	for _, dir := range assetLocations {
 		split := strings.SplitAfterN(filename, dir+string(filepath.Separator), 2)
 		if len(split) > 1 {
-			return filepath.Join(dir, split[len(split)-1])
+			return filepath.ToSlash(filepath.Join(dir, split[len(split)-1]))
 		}
 	}
-
 	return ""
 }

--- a/kit/file_watcher.go
+++ b/kit/file_watcher.go
@@ -17,14 +17,14 @@ const (
 
 var (
 	assetLocations = []string{
-		"templates/customers/",
-		"assets/",
-		"config/",
-		"layout/",
-		"snippets/",
-		"templates/",
-		"locales/",
-		"sections/",
+		filepath.FromSlash("templates/customers"),
+		"assets",
+		"config",
+		"layout",
+		"snippets",
+		"templates",
+		"locales",
+		"sections",
 	}
 )
 
@@ -67,7 +67,7 @@ func (watcher *FileWatcher) watchDirectory(dir string) error {
 	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
 		if info.IsDir() && !watcher.filter.matchesFilter(path) && path != dir {
 			for _, dir := range assetLocations {
-				if strings.Contains(path+"/", dir) {
+				if strings.HasPrefix(path, dir) {
 					if err := watcher.watcher.Add(path); err != nil {
 						return fmt.Errorf("Could not watch directory %s: %s", path, err)
 					}
@@ -173,9 +173,9 @@ func extractAssetKey(filename string) string {
 	filename = filepath.ToSlash(filename)
 
 	for _, dir := range assetLocations {
-		split := strings.SplitAfterN(filename, dir, 2)
+		split := strings.SplitAfterN(filename, dir+string(filepath.Separator), 2)
 		if len(split) > 1 {
-			return fmt.Sprintf("%s%s", dir, split[len(split)-1])
+			return filepath.Join(dir, split[len(split)-1])
 		}
 	}
 

--- a/kit/file_watcher.go
+++ b/kit/file_watcher.go
@@ -65,6 +65,10 @@ func newFileWatcher(client ThemeClient, dir, notifyFile string, recur bool, filt
 func (watcher *FileWatcher) watchDirectory(root string) error {
 	root = filepath.Clean(root)
 	return filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
 		if info.IsDir() && !watcher.filter.matchesFilter(path) && path != root {
 			for _, dir := range assetLocations {
 				if strings.HasPrefix(path, filepath.Join(root, dir, string(filepath.Separator))) {

